src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/ConditionAttribute.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit.Abstractions;
+using System.Linq;
+
+namespace Infrastructure.Common
+{
+    // The [Condition] attribute can be applied to any test method that
+    // is marked with [WcfFact] or [WcfTheory].  It provides zero or more
+    // member names to be evaluated at runtime to determine whether the
+    // test should be run or skipped
+    // Examples: 
+    //  [Condition(nameof(Root_Certificate_Installed))]
+    //  [Condition(nameof(Root_Certificate_Installed), nameof(Client_Certificate_Installed))]
+
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+    public sealed class ConditionAttribute : WcfSkippableAttribute
+    {
+        public string[] Conditions { get; private set; }
+
+        public ConditionAttribute(params string[] conditions)
+        {
+            Conditions = conditions;
+        }
+
+        public override string GetSkipReason(ITestMethod testMethod)
+        {
+            // A null or empty list of conditions is treated as "no conditions",
+            // and the test cases will not be skipped.
+            // Example: [Condition()] or [Condition((string[]) null)]
+            int conditionCount = Conditions == null ? 0 : Conditions.Length;
+            if (conditionCount == 0)
+            {
+                return null;
+            }
+
+            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            Type testMethodDeclaringType = testMethodInfo.DeclaringType;
+
+            List<string> falseConditions = new List<string>(conditionCount);
+
+            foreach (string entry in Conditions)
+            {
+                string conditionMemberName = entry;
+
+                // Null condition member names are silently tolerated
+                if (string.IsNullOrWhiteSpace(conditionMemberName))
+                {
+                    continue;
+                }
+
+                string[] symbols = conditionMemberName.Split('.');
+                Type declaringType = testMethodDeclaringType;
+
+                if (symbols.Length == 2)
+                {
+                    conditionMemberName = symbols[1];
+                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
```

Comment:
>FirstOrDefault [](start = 129, length = 14)

SingleOrDefault might be better in case there are multiple matches. I think we would prefer a failure if there's ambiguity rather than it being non-deterministic.